tests/conftest.py

```diff
@@ -1,5 +1,23 @@
+from dataclasses import make_dataclass
+
+import pytest
 from jax import config
 
 
 def pytest_sessionstart(session):  # noqa: ARG001
     config.update("jax_enable_x64", val=True)
+
+
+def _category_class_factory(values: list[int]):
+    init = [(f"cat{i}", int, value) for i, value in enumerate(values)]
+    return make_dataclass("CategoryClass", init)
+
+
+@pytest.fixture(scope="session")
```

Comment:
Just to check whether you really need the two functions here? I.e., with and without underscore? Can't we just decorate the function itself or does that lead to issues because it takes an argument?

Reply:
Yes, the argument becomes a problem. While pytest supports fixtures that have arguments, it seems like you have to parametrize it via the parametrize decorator, and that would not work in our case, because we need to use different versions of it in the same test.